### ISSUE TITLE
Remove unneeded code from twisted.python.dist._checkCPython

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#6942]
|<img alt="Alex's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| Alex reported|
|-|-|
|Trac ID|trac#6942|
|Type|enhancement|
|Created|2014-02-02 17:59:41Z|
|Branch|https://github.com/twisted/twisted/tree/uneeded-cpython-check-6942|

Now that Twisted requires 2.6, there's a whole bunch of dead code there.

Attachments:

* [t6942.diff](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/4b1/4b1fce7cb60b8a37f4355a6163e1a8ce94bad71b/bab9480baee1ced1cf051a29e7b76fe3283d48a3.diff) (4863 bytes) - added by Alex on 2014-02-02 17:59:58Z - 

<details><summary>Searchable metadata</summary>

```
trac-id__6942 6942
type__enhancement enhancement
reporter__Alex Alex
priority__normal normal
milestone__ 
branch__branches_uneeded_cpython_check_6942 branches/uneeded-cpython-check-6942
branch_author__hawkowl hawkowl
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1391363981000000 1391363981000000
changetime__1391446353000000 1391446353000000
version__None None
owner__hawkowl hawkowl

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)|@hawkowl set owner to @hawkowl|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_3"></a>|@hawkowl commented|
|-|-|

(In [41529]) Branching to uneeded-cpython-check-6942.

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_4"></a>|@hawkowl commented|
|-|-|

(In [41530]) Apply t6942.diff

refs [#6942](https://github.com/twisted/twisted/issues/6942)

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_5"></a>|@hawkowl commented|
|-|-|

Buildbots are green, everything looks good, will merge.

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_6"></a>|@hawkowl set status to `closed`|
|-|-|

(In [41532]) Merge uneeded-cpython-check-6942: Remove unneeded code from twisted.python.dist._checkCPython

Author: Alex
Reviewer: hawkowl
Fixes: [#6942](https://github.com/twisted/twisted/issues/6942)

Removes code not required due to Twisted moving to 2.6+.
